lender.hpp

```diff
@@ -151,25 +148,51 @@ class recycle_memory {
          */
         ~recycle_memory();
 
-        /* get a shared pointer to the buffer we want to fill with data 
-         * NOTE: this is a blocking operation until a buffer is free
-         */
-        auto fill() -> buffer_ptr<T>;
-
-        /* give a shared pointer back to be queued for operation 
-         * NOTE: this is a blocking operation
-         */
-        void queue(buffer_ptr<T> ptr);
+};
 
-        /* get a shared pointer from the queue to operate on  
-         * NOTE: this is a blocking operation until a buffer is queued
-         */
-        auto operate() -> buffer_ptr<T>;
+template <typename T>

```

Comment:
AFAIK, the library class definition has to live in the overarching `lender.hpp` to be able to include only one file when using buffer_ptr/library/mailbox otherwise the class itself is not defined when `lender.hpp` is included